polyfills.js

```diff
@@ -94,25 +94,58 @@ function patch (fs) {
     fs.rename = (function (fs$rename) { return function (from, to, cb) {
       var start = Date.now()
       var backoff = 0;
+      var backoffUntil = start + 60000;
       fs$rename(from, to, function CB (er) {
-        if (er
-            && (er.code === "EACCES" || er.code === "EPERM")
-            && Date.now() - start < 60000) {
+        if (er && (er.code === "EACCES" || er.code === "EPERM") && Date.now() < backoffUntil) {
           setTimeout(function() {
-            fs.stat(to, function (stater, st) {
-              if (stater && stater.code === "ENOENT")
-                fs$rename(from, to, CB);
-              else
-                cb(er)
+            fs.stat(from, function (erFrom, statFrom) {
+              if (erFrom) return cb(erFrom)
+              fs.stat(to, function (erTo, statTo) {
+                if (statFrom.size === statTo.size && 
+                  statFrom.ctime === statTo.ctime) {
+                  cb(null)
+                } else
+                  fs$rename(from, to, CB)
+              });
             })
           }, backoff)
-          if (backoff < 100)
+          if (backoff < 250)
             backoff += 10;
-          return;
+        } else if (backoff && er && er.code === "ENOENT") {
+          // The source does no longer exist so we can assume it was moved during one of the tries
+          if (cb) cb(null)
+        } else {
+          if (cb) cb(er)
         }
-        if (cb) cb(er)
       })
     }})(fs.rename)
+
+    fs.renameSync = (function (fs$renameSync) { return function (from, to) {
+      var start = Date.now()
+      var backoff = 0;
+      var backoffUntil = start + 60000;
+      function tryRename () {
+        try {
+          fs$renameSync(from, to)
+        } catch (e) {
+          if ((e.code === "EACCS" || e.code === "EPERM") && start < backoffUntil) {
+            if (backoff < 100)
+              backoff += 10
+            var waitUntil = Date.now() + backoff
+            while (waitUntil > Date.now()){}
+            tryRename()
+          } else if (backoff > 0 && e.code === "ENOENT") {
+            // The source does no longer exist because it was moved during one of the retries
+            // so we can safely ignore this exception
+          } else {
+            throw e
+          }
+          // Wait until destination exists and source no longer exists or that we've reached the backoff limit
+          while ((fs.existsSync(from) || !fs.existsSync(to)) && Date.now() < backoffUntil) {}
```

Comment:
Possibly the worst hack ever, but it seems to be the only real way to solve inconsistent behavior on Windows where the destination and source can exist at the same time.